subsys/pm/pm.c

```diff
@@ -117,12 +140,22 @@ bool pm_state_force(uint8_t cpu, const struct pm_state_info *info)
 	return true;
 }
 
-bool pm_system_suspend(int32_t ticks)
+bool pm_system_suspend(int32_t kernel_ticks)
 {
 	uint8_t id = _current_cpu->id;
 	k_spinlock_key_t key;
+	int32_t ticks = kernel_ticks, events_ticks;
+
+	SYS_PORT_TRACING_FUNC_ENTER(pm, system_suspend, kernel_ticks);
 
-	SYS_PORT_TRACING_FUNC_ENTER(pm, system_suspend, ticks);
+	/*
+	 * CPU needs to be fully wake up before the event is triggered.
+	 * We need to find out first the ticks to the next event
+	 */
+	events_ticks = pm_policy_next_event_ticks();
+	if (events_ticks >= 0) {
+		ticks = ticks_expiring_sooner(kernel_ticks, events_ticks);
+	}
```

Comment:
This condition is already checked inside `ticks_expiring_sooner()`.
```suggestion
	ticks = ticks_expiring_sooner(kernel_ticks, events_ticks);
```

With this change, line 147 can be simplified as well:
```
int32_t ticks, events_ticks;
```